test-basic/documents-data-movement-transformAll.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const should = require('should');
+const testconfig = require('../etc/test-config.js');
+const marklogic = require('../');
+const dbWriter = marklogic.createDatabaseClient(testconfig.restWriterConnection);
+const restAdminDB = marklogic.createDatabaseClient(testconfig.restAdminConnection);
+const Stream = require('stream');
+const streamToArray = require('stream-to-array');
+const fs = require('fs');
+const expect = require('chai').expect;
+
+let uriStream = new Stream.PassThrough({objectMode: true});
+let urisList = [];
+let result = new Set();
+let summaryValue = null;
+let categoriesUrisList = [];
+let xqyTransformName = 'flagParam';
+let xqyTransformPath = './test-basic/data/flagTransform.xqy';
+let readable = new Stream.Readable({objectMode: true});
+let transformStream = new Stream.PassThrough({objectMode: true});
+let uris = [];
```

Comment:
If these variables are only needed inside the describe(), it would be better to declare them inside the describe()?

Minimizing scope is a good habit to cultivate even in tests, because it can minimize unexpected interactions.